packages/devtools_app/lib/src/flutter/app.dart

```diff
@@ -43,16 +35,33 @@ class DevToolsApp extends StatefulWidget {
 // TODO(https://github.com/flutter/devtools/issues/1146): Introduce tests that
 // navigate the full app.
 class DevToolsAppState extends State<DevToolsApp> {
+  ThemeData theme;
+
   @override
   void initState() {
     super.initState();
+    theme = themeFor(isDarkTheme: devtools_theme.isDarkTheme);
   }
 
   /// Generates routes, separating the path from URL query parameters.
   Route _generateRoute(RouteSettings settings) {
     final uri = Uri.parse(settings.name);
     final path = uri.path;
 
+    // Update the theme based on the query parameters.
+    WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
why are we setting the theme after the frame? that seems a frame too late.

Reply:
We can't setState during a build.

This is an update to the theme, so it's OK if it happens one frame later.

Reply:
can you add a TODO indicating we would prefer to update this a frame earlier but we don't see an obvious way to do that?

Reply:
It's actually fairly tractable to do that way with a NavigatorObserver, there's just more boilerplate (we need to update the route for every possible route transition).

I've left a TODO.

Note that Flutter animates the theme change, so the transition will take a decent fraction of a second.